### ISSUE TITLE
in an HA cluster, bringing down one of the 'controlplane' nodes caused a 'worker' node to become unavailable.

### ISSUE THREAD

**Issue by sdanieru** [rancher/rancher#16497]
**Bug**
With a single rancher installation, I created a k8s cluster with 3 'controlplane,etcd' nodes, and 2 'worker' nodes. To test resiliency, I've been bringing down one 'controlplane,etcd' node at a time to see how the cluster responds. When I brought down 'controlplane, etcd' node A , the rancher UI reported that 'controlplane,etcd' node A was unavailable (expected), but also reported that worker A had become unavailable! - 'Kubelet stopped posting node status'.

**Steps to reproduce (least amount of steps as possible):**
1. Via Rancher UI, create a cluster with three nodes with 'controlplane,etcd' roles.
2. Via Rancher UI, create 2 or more 'worker' nodes.
3. bring down only 'controlplane, etcd' node A; observe cluster and node status
4. repeat for 'controlplane,etc' nodes B and C
**Result:**
Bringing down 'controlplane,etc' node A, rendered 'worker' node A unavailable. Bringing down 'controlplane,etcd' nodes B and C did not affect any worker nodes.

**other useful info**
Some of the hosts are running Docker 18.6.1, others 17.3.3

**Environment information**
- Rancher version 2.0.8
- Installation option: Rancher is single, cluster is HA

**Cluster information**
- Cluster type: Custom, Ubuntu 16.04 on hosts
- Machine types: all metal with ranges of 4 to 12 cores, and 16 to 64GB RAM
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GitCommit:"b1b29978270dc22fecc592ac55d903350454310a", GitTreeState:"clean", BuildDate:"2018-07-17T18:53:20Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-07T23:08:19Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker (use `docker info`):

```
Containers: 26
 Running: 15
 Paused: 0
 Stopped: 11
Images: 161
Server Version: 17.03.3-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 6c463891b1ad274d505ae3bb738e530d1df2b3c7
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-138-generic
Operating System: Ubuntu 16.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 12
Total Memory: 62.81 GiB
Name: kdev03-a
ID: XD7E:KYRZ:BQU3:ZZAJ:DRUL:SKXE:I74X:AI6R:2IT6:NAAW:QNZ6:DWZI
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

```

**Comment by cjellick** [comment]
Did the worked node come back to available eventually or did it stay unavailable permanently?

**Comment by sdanieru** [comment]
I didn't leave it in that state for longer than 30m so I can only say the worker didn't recover during that time, at which point I manually intervened by bringing the control node up again, at which time the worker came up again as well.
It's worth noting that I have since upgraded rancher to 2.1.1 (I was on 2.0.8), and the k8s cluster to 1.11.3. I then ran through the same test scenarios and had absolutely no problems - the system gracefully handled the case of 1 of the three control nodes being down: no workers failed; no services failed. I was particularly happy to see that in the case of the control node going down that rancher was using for cluster control, after rancher briefly complained it couldn't talk to the cluster, it then re-established control (I would assume redirected to an API server on one of the other control nodes)
I remember reading somewhere that for k8s HA, the API servers across control nodes needed to be fronted by a load-balancer. Great to see that Rancher seems to be handling this for me.
